### ISSUE TITLE
Endless loop when loading a freeformat mpeg stream

### ISSUE THREAD

**Issue by eblanca** [DeaDBeeF-Player/deadbeef#2253]
### Steps to reproduce the problem
Open an mpeg audio stream which was encoded as "freeformat"

### What's going on? Describe the problem in as much detail as possible.
Deadbeef loops after opening the "Adding files" dialog.

### Information about the software:
Deadbeef version:
deadbeef master branch, commit 1f833ddbc64fc17d0236df7bc9daa3ddb2bbfe65 (latest)
built with both libmpg123 and libmad mpeg backends, happens using both.
OS: LMDE 3 x64 (Debian based), gcc-6.3.0, binutils-2.28

I added some tiny trace and I see the code loops into mp3parser.c always calling "_parse_packet".
I think this happens because at
https://github.com/DeaDBeeF-Player/deadbeef/blob/1f833ddbc64fc17d0236df7bc9daa3ddb2bbfe65/plugins/mp3/mp3parser.c#L426
the code checks for res==-1 (as _parse_packet returns when finding freeformat frames) but this is a somewhat "special" case.
Other players (mpg123, madplay) handle those streams just fine.

**Comment by Alexey-Yakovenko** [comment]
Please provide some extra information, about how to produce such stream.

**Comment by eblanca** [comment]
run lame with:
`lame -b 200 --freeformat input.wav output.mp3`
(any bitrate value will work anyway, lame supports up to 640 kbps)

**Comment by eblanca** [comment]
Those streams have 0 as bitrate index into their mpeg header.

**Comment by Alexey-Yakovenko** [comment]
OK I can see that deadbeef can't handle this type of files. I never encountered them before. Do you know if there's any documentation about them? File format spec?

**Comment by Alexey-Yakovenko** [comment]
Also, I can't get deadbeef to go into infinite loop. It just takes a long time to parse the stream, as it attempts to interpret it as a regular mp3, but of course that doesn't go well, and it moves one byte at a time through a lot of data trying to sync. It finishes eventually, I guess depends on the CPU speed.

**Comment by Alexey-Yakovenko** [comment]
I suppose the fix you're getting is not quite what you might have expected, but deadbeef currently doesn't support this format, so now it will get detected and ignored.
If you want to actually be able to play this format -- feel free to submit a feature request.
I doubt this will happen any time soon though. It's not a simple change, and seems like a lot of work.

**Comment by eblanca** [comment]
Oh, I love deadbeed the way it is and nothing will change this. Honestly, I don't think supporting these free format streams will make it a "killer application" as very few users know about their existence and those using them are far less.

I just tried feeding deadbeef with one and see what happened.

The detection into my utility mp3guessenc simply seeks for a subsequent mpeg sync bit sequence after finding a freeformat frame, as the header itself can tell nothing about frame length (then, once you know the size of the first frame, everything becomes easier). It's a lot of work, I agree. Further, not much documentation about them so, after reading some mpeg related article I had to pick some and look into using a hex viewer. Go figure.

Thank you.
